helm/kong-app/templates/_gs-helpers.tpl

```diff
@@ -48,3 +48,39 @@ app.kubernetes.io/name: {{ include "kong.chart-name" . | quote }}
 app.kubernetes.io/component: app
 app.kubernetes.io/instance: "{{ .Release.Name }}"
 {{- end -}}
+
+{{/*
+Create a custom helper for omitting seccompProfile from securityContexts in pre-upgrade jobs
+Will be removed by the next sync from upstream
+THIS IS OKAY!
```

Comment:
There is some nifty helm stuff happening here!

Mostly just for my own understanding -- why is it necessary to support disabling the seccomp profile?
Also -- `THIS IS OKAY!` -- do you have some automation that puts it back? Or you just manually re-add it in the future?

Reply:
> why is it necessary to support disabling the seccomp profile?

This is required because there is a helm pre-upgrade hook which creates a Job. The way helm works is that hook templates are rendered before the rest of the chart. I'e encountered the problem that the pre-upgrade job could not finish because kubernetes was complaining that usage of `seccompProfile` is not allowed because the required annotation on the PSP is applied **after** the Job in the pre-upgrade hook requires the annotation to be in place.

> do you have some automation that puts it back?

Yes, this repository is being synchronized with the upstream chart github.com/Kong/charts using vendir. The change is not in our fork, so the original will get removed with the next sync.

Reply:
Very interesting. Thanks!